checkpoint/resource_checkpoint_management_access_rule.go

```diff
@@ -1069,7 +1069,7 @@ func updateManagementAccessRule(d *schema.ResourceData, m interface{}) error {
 				if d.HasChange("user_check.0.frequency") {
 					userCheckPayload["frequency"] = d.Get("user_check.0.frequency").(bool)
 				}
-				if d.HasChange("user_check.0.web_server") {
+				if d.HasChange("user_check.0.interaction") {
 					userCheckPayload["interaction"] = d.Get("user_check.0.interaction").(bool)
```

Comment:
invalid casting

Reply:
Sorry about that, I updated the type casting and ran make fit, and make test.  I'm still new to go and trying to familiarize myself with the project.  Do you have any additional instructions on the make testacc other than what's in the readme?

Reply:
That's ok. Thanks for the change.
You don't need to add casting for `every` field it was looking good.
You can test locally by creating / updating access rule with `user_check` field and make sure to define all properties: `interaction`, `custom_frequency` , `frequency` and `confirm`.

Reply:
Finally figured out how to install a custom provider.  I'll get it tested and report back.

Reply:
```
resource "checkpoint_management_access_rule" "rule4" {
  layer = "Test-Layer"
  position = {top = "top"}
  name = "test4"
  source = ["Any"]
  destination = ["Any"]
  service = ["Any"]
  track = {
    type = "Log"
    accounting = false
    alert = "none"
    enable_firewall_session = false
    per_connection = true
    per_session = false
  }
  user_check  {
    interaction = "Blocked Message - Access Control"
 }
  enabled = true
}
```
![image](https://user-images.githubusercontent.com/67339472/219828964-e4678be8-76ee-4469-9f86-cbccca1dbd4b.png)

```
resource "checkpoint_management_access_rule" "rule5" {
  layer = "Test-Layer"
  position = {below = checkpoint_management_access_rule.rule4.name}
  name = "test5"
  source = ["Any"]
  destination = ["Any"]
  service = ["Any"]
  action = "Ask"
  track = {
    type = "Log"
    accounting = false
    alert = "none"
    enable_firewall_session = false
    per_connection = true
    per_session = false
  }
  user_check  {
    interaction = "Company Policy"
    frequency = "Once a day"
    confirm = "Per application/site"
 }
  enabled = true
}

resource "checkpoint_management_access_rule" "rule6" {
  layer = "Test-Layer"
  position = {below = checkpoint_management_access_rule.rule5.name}
  name = "test5"
  source = ["Any"]
  destination = ["Any"]
  service = ["Any"]
  action = "Ask"
  track = {
    type = "Log"
    accounting = false
    alert = "none"
    enable_firewall_session = false
    per_connection = true
    per_session = false
  }
  user_check  {
    interaction = "Company Policy"
    custom_frequency {
      every = "1"
      unit = "hours"
    }
    confirm = "Per application/site"
 }
  enabled = true
}
```
![image](https://user-images.githubusercontent.com/67339472/219829622-75d41146-3135-4ee9-96b3-add3cbfee307.png)

Reply:
Looks great! Thank you @bmyers-dev.
Can you include only the changes relevant for this issue without the other formatting? I prefer to merge only the required changes for this fix.